pkg/ipcache/ipcache.go

```diff
@@ -162,16 +163,38 @@ func (ipc *IPCache) GetK8sMetadata(ip string) *K8sMetadata {
 	return nil
 }
 
-func (ipc *IPCache) updateNamedPorts() (namedPortsChanged bool) {
+func (ipc *IPCache) updateNamedPorts(newIP string, newNamedPorts policy.NamedPortsMap) (namedPortsChanged bool) {
 	// Collect new named Ports
 	npm := make(policy.NamedPortsMap, len(ipc.namedPorts))
-	for _, km := range ipc.ipToK8sMetadata {
+	for ip, km := range ipc.ipToK8sMetadata {
 		for name, value := range km.NamedPorts {
-			if oldValue, done := npm[name]; done {
-				// Do not process duplicate values
-				if oldValue != value {
-					log.Warning("Confilicting named port definition")
+			if ip != newIP {
+				// Check if there is a conflict between any old named port from other IPs
+				// with any of the newly added (or changed) values
+				if newValue, exists := newNamedPorts[name]; exists && newValue != value {
+					scopedLog := log.WithFields(logrus.Fields{
+						logfields.IPAddr:   newIP,
+						logfields.Identity: ipc.ipToIdentityCache[newIP],
+					})
+					newMeta := ipc.ipToK8sMetadata[newIP]
+					scopedLog = scopedLog.WithFields(logrus.Fields{
+						logfields.K8sPodName:   newMeta.PodName,
+						logfields.K8sNamespace: newMeta.Namespace,
+						logfields.NamedPorts:   newMeta.NamedPorts,
+					})
+					oldMeta := ipc.ipToK8sMetadata[ip]
+					scopedLog = scopedLog.WithFields(logrus.Fields{
+						"other" + logfields.K8sPodName:   oldMeta.PodName,
+						"other" + logfields.K8sNamespace: oldMeta.Namespace,
+						"other" + logfields.NamedPorts:   oldMeta.NamedPorts,
+					})
+					scopedLog.Warningf("Confilicting named port definition: %s: %d/%s != %d/%s", name,
```

Comment:
nit: `s/Confilicting/Conflicting`

ping me on slack once this is fixed to merge it as we don't need to run the CI again since it was all green.

Reply:
Fixed, thanks!